pubsublite/config.go

```diff
@@ -211,6 +212,121 @@ func (tc *TopicConfigToUpdate) toUpdateRequest() *pb.UpdateTopicRequest {
 	}
 }
 
+// ExportDestinationConfig is the configuration for exporting to a destination.
+// Implemented by *PubSubConfig.
+type ExportDestinationConfig interface {
+	setExportConfig(ec *pb.ExportConfig) string
+}
+
+// PubSubConfig configures messages to be exported to a Pub/Sub topic.
+// Implements the ExportDestinationConfig interface.
+type PubSubConfig struct {
+	// The path of a Pub/Sub topic, in the format:
+	// "projects/PROJECT_ID/topics/TOPIC_ID".
+	Topic string
+}
+
+func (pc *PubSubConfig) setExportConfig(ec *pb.ExportConfig) string {
+	ec.Destination = &pb.ExportConfig_PubsubConfig{
+		PubsubConfig: &pb.ExportConfig_PubSubConfig{Topic: pc.Topic},
+	}
+	return "export_config.pubsub_config"
+}
+
+// ExportState specifies the desired state of an export subscription.
+type ExportState int
+
+const (
+	// UnspecifiedExportState represents an unset export state.
+	UnspecifiedExportState ExportState = iota
+
+	// ExportActive specifies that export processing should be enabled.
+	ExportActive
+
+	// ExportPaused specifies that export processing should be suspended.
+	ExportPaused
+)
+
+// Status references google.golang.org/genproto/googleapis/rpc/status.
+type Status = status.Status
```

Comment:
Do you actually care about the whole status or just the code from the status? If it is just the code you could alias that, or maybe even just use an in and map it?

Reply:
The entire status proto may be useful for debugging, as the message can help differentiate between errors with the same code. The error details may also be populated with metadata.

I used the Status proto, for consistency with bigtable: https://pkg.go.dev/cloud.google.com/go/bigtable#EncryptionInfo

However, I was unsure whether it would be better to convert to https://pkg.go.dev/google.golang.org/grpc/status#Status.

Reply:
Obsolete. We changed the API to return enum values.